contracts/hardhat.config.js

```diff
@@ -0,0 +1,23 @@
+require("@nomiclabs/hardhat-waffle");
+
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+module.exports = {
+  solidity: "0.8.4",
+  paths: {
+    artifacts: "./artifacts",
+    cache: "./cache",
+    sources: "./contracts",
+    tests: "./test",
+  },
+  overrides: {
+    "contracts/v0.8/*": {
```

Comment:
not sure if this is the way we want to set it up

Reply:
Consider checking out how this is configured in the [LinkToken repo](https://github.com/smartcontractkit/LinkToken/blob/master/hardhat.config.ts). There we have a complex Hardhat situation with multiple versions (0.4, 0.6, 0.7) and multiple targets (EVM, OVM). We'll soon need to bring it here to support Optimism.